### PR TITLE
feat(yabloc): add camera and vector map localization

### PR DESCRIPTION
**Pull request by KYabuuchi**
## Description

Add [YabLoc (localization with camera and vector map)](https://github.com/tier4/YabLoc) as a new location estimation method.
The user can choose whether to run NDT or YabLoc by specifying `localization_mode` as an argument when launching autoware.

The following figure shows where the code has been changed and added. Red frames indicate edited files, and red blocks indicate added files.
image

## Related links

* https://github.com/tier4/YabLoc
* https://github.com/orgs/autowarefoundation/discussions/3484

## Tests performed

Please see https://github.com/autowarefoundation/autoware.universe/pull/3946

## Notes for reviewers

This PR needs to be merged with 
* https://github.com/autowarefoundation/autoware.universe/pull/3946

There is also a PR to autoware-documentation.
* https://github.com/autowarefoundation/autoware-documentation/pull/394 

## Interface changes

Nothing.

## Effects on system behavior

By default, autoware runs as before.
If `localization_mode:=yabloc` is added as an argument when launching autoware, YabLoc runs instead of NDT.

When yabloc is used, `matching_score`, which is NDT's diagnostics, is not published.
[Note that the absence of `matching_score` does not affect the system error monitor.](https://github.com/autowarefoundation/autoware.universe/blob/main/system/system_error_monitor/config/system_error_monitor.param.yaml#L26)

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: autoware_launch/config/localization/yabloc/expressway/graph_segment.param.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+/**:
+  ros__parameters:
+      # graph_node selects a road surface area from around this height
+      target_height_ratio: 0.85
+
+      #
+      target_candidate_box_width: 15
+
+      # graph_segment_node will pickup additional roadlike areas
```
Comment (kminoda) [inline autoware_launch/config/localization/yabloc/expressway/graph_segment.param.yaml:9]: Please fix these spell-check errors
image

(I've added ximgproc here: https://github.com/tier4/autoware-spell-check-dict/pull/542, but for others I think its better to just change them to something else)

**Comment by KYabuuchi** [comment]
@yukkysaito @kminoda @YamatoAndo Please review this pull request when you have time :pray:

**Review by yukkysaito — APPROVED** [review]
LGTM

File: autoware_launch/launch/autoware.launch.xml [hunk 2]
```diff
@@ -38,6 +38,8 @@
   <!-- Perception -->
   <arg name="perception_mode" default="lidar" description="select perception mode. camera_lidar_radar_fusion, camera_lidar_fusion, lidar_radar_fusion, lidar, radar"/>
   <arg name="traffic_light_recognition/enable_fine_detection" default="true" description="enable traffic light fine detection"/>
+  <!-- Localization -->
+  <arg name="localization_mode" default="ndt" description="select localization mode. ndt, yabloc"/>
```
Comment (yukkysaito) [inline autoware_launch/launch/autoware.launch.xml:42]: I would like to make modes like lidar or camera if possible.

Comment (KYabuuchi) [reply autoware_launch/launch/autoware.launch.xml]: I changed mode names in 6d91776. Now the users can launch as `localization_mode:=lidar` or `localization_mode:=camera` 
Is this what you wish?

Comment (yukkysaito) [reply autoware_launch/launch/autoware.launch.xml]: Yes 👍

**Pull request merged**